torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -497,6 +497,7 @@ void AliasDb::analyzeImpl(Node* node) {
     case prim::Function:
     case prim::CreateObject:
     case prim::tolist:
+    case prim::CudaFusionGuard:
       return analyzeCreator(node);
```

Comment:
nitpick: I'm wondering if it might be cleaner to assert that the output type of a guard is `BoolType` and just `return;`? If `CudaFusionGuard` will ever start returning refined tensor types, then creating a fresh alias for it would be incorrect.

Reply:
Good point! Let me update it.

Reply:
Hmmmm. after examine this again, I'm a little concerned at this. If we just return, does that mean the boolean we are returning becomes a wildcard?
I'm leaning towards keeping alias analysis as tight as possible in general. I can put a comment in the interface for the op registration, so if we do change the schema in the future, we would go validate the alias analysis.

BTW, I'm already marking output type in the fusion pass: https://github.com/pytorch/pytorch/pull/45706/files#diff-43281337f2ca4b6fa9c3424d6352cf54R984

Reply:
> Hmmmm. after examine this again, I'm a little concerned at this. If we just return, does that mean the boolean we are returning becomes a wildcard?

Booleans aren't mutable/reference types; they can't alias as far as i know. tagging @eellison to double check.

> if we do change the schema in the future, we would go validate the alias analysis.

I forgot why we couldn't use a schema for this op?

> Is there somewhere I should set the type again in aliasdb?

Running `analyzeSubgraph` should be enough, I think. AA doesn't differentiate between tensor of different shapes/devices if this is what you mean. Shapes/devices is the only additional information your fusion pass adds.

Reply:
Oops, I think I understand it now. I wasn't aware of the distinction between a scalar type from a tensor type.
The only reason that we didn't use a schema for this op is my poor knowledge of AA. I'll push an update soon~ish

Reply:
Here we go:
https://github.com/pytorch/pytorch/pull/45706/commits/e6819ad4c4ca82a364a66d2398a1fda28988f3b0